sdk/identity/identity/src/credentials/azureCliCredential.ts

```diff
@@ -68,59 +78,58 @@ export class AzureCliCredential implements TokenCredential {
-              );
-              logger.getToken.info(formatError(scopes, error));
-              throw error;
-            } else if (isLoginError) {
-              const error = new CredentialUnavailableError(
-                "Please run 'az login' from a command prompt to authenticate before using this credential."
-              );
-              logger.getToken.info(formatError(scopes, error));
-              throw error;
-            }
-            const error = new CredentialUnavailableError(obj.stderr);
-            logger.getToken.info(formatError(scopes, error));
-            throw error;
-          } else {
-            responseData = obj.stdout;
-            const response: { accessToken: string; expiresOn: string } = JSON.parse(responseData);
-            logger.getToken.info(formatSuccess(scopes));
-            const returnValue = {
-              token: response.accessToken,
-              expiresOnTimestamp: new Date(response.expiresOn).getTime()
-            };
-            resolve(returnValue);
-            return returnValue;
-          }
-        })
-        .catch((err) => {
-          span.setStatus({
-            code: SpanStatusCode.ERROR,
-            message: err.message
-          });
-          logger.getToken.info(formatError(scopes, err));
-          reject(err);
-        });
-    });
+    const { span } = createSpan("AzureCliCredential-getToken", options);
+
+    try {
+      const obj = await cliCredentialInternals.getAzureCliAccessToken(resource);
+      if (obj.stderr) {
+        const isLoginError = obj.stderr.match("(.*)az login(.*)");
+        const isNotInstallError =
+          obj.stderr.match("az:(.*)not found") || obj.stderr.startsWith("'az' is not recognized");
+        if (isNotInstallError) {
+          const error = new CredentialUnavailableError(
+            "Azure CLI could not be found.  Please visit https://aka.ms/azure-cli for installation instructions and then, once installed, authenticate to your Azure account using 'az login'."
+          );
+          logger.getToken.info(formatError(scopes, error));
+          throw error;
+        } else if (isLoginError) {
+          const error = new CredentialUnavailableError(
+            "Please run 'az login' from a command prompt to authenticate before using this credential."
+          );
+          logger.getToken.info(formatError(scopes, error));
+          throw error;
+        }
+        const error = new CredentialUnavailableError(obj.stderr);
+        logger.getToken.info(formatError(scopes, error));
+        throw error;
+      } else {
+        responseData = obj.stdout;
+        const response: { accessToken: string; expiresOn: string } = JSON.parse(responseData);
+        logger.getToken.info(formatSuccess(scopes));
+        const returnValue = {
+          token: response.accessToken,
+          expiresOnTimestamp: new Date(response.expiresOn).getTime()
+        };
+        return returnValue;
+      }
+    } catch (err) {
```

Comment:
Thanks to @willmtemple for noticing the weird promise here. I changed it, though I’m tempted, and I’m resisting, from doing further cleaning 🤔